ddtrace/internal/_encoding.pyx

```diff
@@ -509,7 +510,7 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
     cpdef flush(self):
         raise NotImplementedError()
 
-    cdef int pack_span(self, object span, void *dd_origin) except? -1:
+    cdef int pack_span(self, object span, void *dd_origin, int i) except? -1:
```

Comment:
I wonder if having something more explicit like `bool is_first_span` or something

Or maybe, we can just change `i` to be a better name like `index_in_chunk` or... idk, naming is hard.